ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -248,15 +258,92 @@ impl Blockstore {
             return Err(e);
         }
         write_timer.stop();
+
+        let mut purge_files_in_range_timer = Measure::start("delete_file_in_range");
+        // When the from_slot is 0, it is safe and more effective to run purge_files_in_range.
+        // It is safe because if purge_files_in_range delete any sst file that contains any
+        // range-deletion tombstone, the deletion range of that tombstone will be completely
+        // covered by the new range-delete tombstone (0, to_slot) issued above.
+        // It is more effective because we are more likely to delete sst files which key
+        // ranges are completely older than to_slot.
```

Comment:
Here're more detailed reasons why we only want to purge files when `from_slot` is 0.